models/surveys.go

```diff
@@ -486,13 +493,42 @@ func (api *API) Info(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-// AllSurveys returns summaries of all known surveys. The surveys are returned in ascending short name order.
+//AllSurveys handler returns a list of all surveys
 func (api *API) AllSurveys(w http.ResponseWriter, r *http.Request) {
-	rows, err := api.AllSurveysStmt.Query()
+	api.GetSurveys(w, r, "")
+}
 
-	if err != nil {
-		http.Error(w, "AllSurveys query failed", http.StatusInternalServerError)
-		return
+//SurveysByType handler returns a list of all surveys of a particular type e.g Business or Social
+func (api *API) SurveysByType(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	surveyType := vars["surveyType"]
```

Comment:
Would it be sensible to check that the supplied request is one of the 3 expected values, and return http error 400 bad request if it's not one of them? Also, is the DB set to be case-sensitive? (i.e. will "BUSINESS" return zero surveys?)